lib/widgets/button.dart

```diff
@@ -0,0 +1,169 @@
+          WidgetState.pressed: designVariables.btnBgAttHighIntInfoActive,
+          ~WidgetState.pressed: designVariables.btnBgAttHighIntInfoNormal,
+        });
+    }
+  }
+
+  Color _labelColor(DesignVariables designVariables) {
+    switch ((attention, intent)) {
+      case (ZulipWebUiKitButtonAttention.medium, ZulipWebUiKitButtonIntent.info):
+        return designVariables.btnLabelAttMediumIntInfo;
+      case (ZulipWebUiKitButtonAttention.high, ZulipWebUiKitButtonIntent.info):
+        return designVariables.btnLabelAttHigh;
+    }
+  }
+
+  TextStyle _labelStyle(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    return TextStyle(
+      color: _labelColor(designVariables),
+      fontSize: 17,
+      height: 1.20,
+      letterSpacing: proportionalLetterSpacing(context,
+        0.006, baseFontSize: 17),
+    ).merge(weightVariableTextStyle(context, wght: 600));
+  }
+
+  BorderSide _borderSide(DesignVariables designVariables) {
+    switch (attention) {
+      case ZulipWebUiKitButtonAttention.medium:
+        // TODO inner shadow effect like `box-shadow: inset`, following Figma;
+        //   needs Flutter support for something like that:
+        //     https://github.com/flutter/flutter/issues/18636
+        //     https://github.com/flutter/flutter/issues/52999
+        //   For now, we just use a solid-stroke border with half the opacity
+        //   and half the width.
+        return BorderSide(
+          color: designVariables.btnShadowAttMed.withFadedAlpha(0.5),
+          width: 0.5);
+      case ZulipWebUiKitButtonAttention.high:
+        return BorderSide.none;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return AnimatedScaleOnTap(
+      scaleEnd: 0.96,
+      duration: Duration(milliseconds: 50),
+      child: TextButton(
+        style: TextButton.styleFrom(
+          padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
+          foregroundColor: _labelColor(designVariables),
+          shape: RoundedRectangleBorder(
+            side: _borderSide(designVariables),
+            borderRadius: BorderRadius.circular(4)),
+          splashFactory: NoSplash.splashFactory,
+          tapTargetSize: MaterialTapTargetSize.shrinkWrap,
+          minimumSize: Size(kMinInteractiveDimension, 28),
+        ).copyWith(backgroundColor: _backgroundColor(designVariables)),
+        onPressed: onPressed,
+        child: ConstrainedBox(
+          constraints: BoxConstraints(maxWidth: 240),
+          child: Text(label,
+            maxLines: 1,
+            style: _labelStyle(context),
+            textAlign: TextAlign.center,
+            overflow: TextOverflow.ellipsis))));
+  }
+}
+
+enum ZulipWebUiKitButtonAttention {
+  high,
+  medium,
+  // low,
+}
+
+enum ZulipWebUiKitButtonIntent {
+  // neutral,
+  // warning,
+  // danger,
+  info,
+  // success,
+  // brand,
```

Comment:
One good reason to add these only lazily, as you're doing here: to keep down how many different colors we have to maintain in theme.dart, until we finish #831.